lib/registration.pm

```diff
@@ -195,27 +195,22 @@ sub add_suseconnect_product {
 
 =head2 ssh_add_suseconnect_product
 
-    ssh_add_suseconnect_product($remote, $name, [$version, [$arch, [$params]]]);
+    ssh_add_suseconnect_product($remote, $name, [$version, [$arch, [$params, [$timeout, [$retries, [$delay]]]]]]);
 
 Wrapper for SUSEConnect -p $name  over ssh.
 =cut
 sub ssh_add_suseconnect_product {
-    my ($remote, $name, $version, $arch, $params, $timeout, $retry) = @_;
+    my ($remote, $name, $version, $arch, $params, $timeout, $retries, $delay) = @_;
     assert_script_run "sftp $remote:/etc/os-release /tmp/os-release";
     assert_script_run 'source /tmp/os-release';
     $version //= '${VERSION_ID}';
     $arch    //= '${CPU}';
     $params  //= '';
-    $retry   //= 0;                 # run SUSEConnect a 2nd time to workaround the gpg error due to missing repo key on 1st run
     $timeout //= 300;
+    $retries //= 3;
+    $delay   //= 10;
 
-    my $result = script_run("ssh $remote sudo SUSEConnect -p $name/$version/$arch $params", $timeout);
-    if ($result != 0 && $retry) {
-        if ($name =~ /PackageHub/) {
-            record_soft_failure 'bsc#1124318 - Fail to get module repo metadata - running the command again as a workaround';
-        }
-        assert_script_run("ssh $remote sudo SUSEConnect -p $name/$version/$arch $params", $timeout);
-    }
+    script_retry("ssh $remote sudo SUSEConnect -p $name/$version/$arch $params", delay => $delay, retry => $retries, timeout => $timeout);
```

Comment:
Now the code looks good, I'm starting to think what it mean to do a `SUSEConnect` just twice. Should we consider to make some stats about such failures. And if it happen "always" we should raise a hand? 
For me this question is actually out of the skope of this PR, cause we need to make thinks testable. But lets discuss it in next sync meeting, wdyt?

Reply:
The reason are occasional failures because a ton of jobs get scheduled at the same time and this puts a huge load on the scc servers. Those can be mitigated by running `SUSEConnect` multiple times with adequate timeout and delay settings and this is the whole reason for this PR.

But this is indeed perhaps food for another meeting :+1: